src/rascal/structure_managers/structure_manager_centers.hh

```diff
@@ -304,6 +306,11 @@ namespace rascal {
       return this->atoms_object;
     }
 
+    bool are_any_centers_masked() const {
+      // center_atoms_mask is true is no center are masked
+      return (not this->get_center_atoms_mask().all());
```

Comment:
Yes, here it is instead of the evaluation - which is possibly a very big number set it at construction.
I am not sure if it is necessary and how often it is evaluated, just a thought.